numpy/lib/arraysetops.py

```diff
@@ -332,14 +338,26 @@ def intersect1d(ar1, ar2, assume_unique=False):
     >>> from functools import reduce
     >>> reduce(np.intersect1d, ([1, 3, 4, 3], [3, 1, 2, 1], [6, 3, 4, 2]))
     array([3])
+    
+    To return the indices of the values common to the input arrays
+    along with the intersected values:
+    
+    >>>np.intersect1d([1,2,3,4],[2,1,4,6],return_indices=True)
+    (array(1,2,4),array([0,1,3]),array([1,0,2]))
     """
     if not assume_unique:
         # Might be faster than unique( intersect1d( ar1, ar2 ) )?
         ar1 = unique(ar1)
         ar2 = unique(ar2)
     aux = np.concatenate((ar1, ar2))
     aux.sort()
-    return aux[:-1][aux[1:] == aux[:-1]]
+    int1d = aux[:-1][aux[1:] == aux[:-1]]
+
+    if return_indices:
+      a_inds, = np.in1d(ar1,int1d).nonzero()
+      b_inds, = np.in1d(ar2,int1d).nonzero()
+      return int1d,a_inds,b_inds
```

Comment:
There is no `a` and `b` referenced elsewhere in this function, better call these `inds1` and `inds2`

Reply:
Or, if you ask me, `indices1` and `indices2`...